### PR TITLE
docs: Add info on GitHub filter push down

### LINKED ISSUE #435: Add documentation on GitHub connector filter push down
Add information about which columns support filter push down in which `github_query_mode`s.

### LINKED ISSUE #436: Update GitHub pulls connector login column
The column has been renamed to `author`

### PR DESCRIPTION
**Pull request by peasee**
## 🗣 Description

* Adds more info on GitHub filter push down, the supported modes, and behaviors.

## 🔨 Related Issues

* Closes #436 
* Closes #435 
* Part of https://github.com/spiceai/spiceai/issues/2832

### CODE REVIEW

File: spiceaidocs/docs/components/data-connectors/github.md [hunk 1]
```diff
@@ -15,8 +15,39 @@ The GitHub data connector can be configured by providing the following `params`.
 - `owner` - Required. Specifies the owner of the GitHub repository.
 - `repo` - Required. Specifies the name of the GitHub repository.
 
+### Filter Push Down
+
+Querying GitHub supports a `github_query_mode` parameter value of either `auto` or `search` in the following connectors:
+
+- Issues; defaults to `auto`, query filters are only pushed down in `search` mode.
+- Pull Requests; defaults to `auto`, query filters are only pushed down in `search` mode.
```
Comment (lukekim) [inline spiceaidocs/docs/components/data-connectors/github.md]:
```suggestion
GitHub queries support a `github_query_mode` parameter, which can be set to either `auto` or `search` for the following types:

- **Issues**: Defaults to `auto`. Query filters are only pushed down to the GitHub API in `search` mode.
- **Pull Requests**: Defaults to `auto`. Query filters are only pushed down to the GitHub API in `search` mode.
```

File: spiceaidocs/docs/components/data-connectors/github.md [hunk 2]
```diff
@@ -15,8 +15,39 @@ The GitHub data connector can be configured by providing the following `params`.
...
+
+The Commits connector supports only `auto` mode, with query filter push down enabled for the `committed_date` column. `commited_date` supports exact matches, or greater/less than matches for dates provided in [ISO8601](https://www.iso.org/iso-8601-date-and-time-format.html) format, like `WHERE committed_date > '2024-09-24'`.
```
Comment (lukekim) [inline spiceaidocs/docs/components/data-connectors/github.md]:
```suggestion
Commits only supports `auto` mode. Query with filter push down is only enabled for the `committed_date` column. `commited_date` supports exact matches, or greater/less than matches for dates provided in [ISO8601](https://www.iso.org/iso-8601-date-and-time-format.html) format, like `WHERE committed_date > '2024-09-24'`.
```

File: spiceaidocs/docs/components/data-connectors/github.md [hunk 3]
```diff
@@ -15,8 +15,39 @@ The GitHub data connector can be configured by providing the following `params`.
...
+
+- GitHub has a limitation in the Search API where it may return more stale data than the standard API used in the default query mode.
+
+:::
+
 ### Querying GitHub Files
 
+:::warning[Limitations]
+
+- `content` column is included only when acceleration is enabled.
```
Comment (lukekim) [inline spiceaidocs/docs/components/data-connectors/github.md]:
```suggestion
- `content` column is fetched only when acceleration is enabled.
```

**Review by lukekim — DISMISSED** [review]

File: spiceaidocs/docs/components/data-connectors/github.md [hunk 4]
```diff
@@ -15,8 +15,39 @@ The GitHub data connector can be configured by providing the following `params`.
 - `owner` - Required. Specifies the owner of the GitHub repository.
 - `repo` - Required. Specifies the name of the GitHub repository.
 
+### Filter Push Down
+
+GitHub queries support a `github_query_mode` parameter, which can be set to either `auto` or `search` for the following types:
+
+- **Issues**: Defaults to `auto`. Query filters are only pushed down to the GitHub API in `search` mode.
```
Comment (Jeadie) [inline spiceaidocs/docs/components/data-connectors/github.md:22]: nit: maybe this should be a table.

**Review by Jeadie — APPROVED** [review]

**Pull request merged**